src/functionapp/ai_ocr/genai_operations.py

```diff
@@ -105,28 +100,32 @@ def perform_gpt_evaluation_and_enrichment(images: list, extracted_data: dict, js
     Here is the JSON schema template that was used for the extraction:
 
     {json_schema}
-    
-    ------
-    
-    {parser.get_format_instructions()}
     """
 
-    chat_template = ChatPromptTemplate.from_messages(
-        [
-            SystemMessage(content=system_message),
-            HumanMessagePromptTemplate.from_template("Here is the extracted data :\n{extracted}"),
-        ]
-    )
-    messages = chat_template.format_messages(extracted=json.dumps(extracted_data, indent=2))
+    chat_history = ChatHistory(system_message = system_message)
+    chat_history.add_user_message(f"Here is the extracted data :\n{json.dumps(extracted_data, indent=2)}")
 
     if images:
-        messages.append(HumanMessage(content="Here are the images from the document:"))
+        chat_history.add_user_message("Here are the images from the document:")
         for img in images:
-            messages.append(HumanMessage(content=[{"type": "image_url", "image_url": {"url": f"data:image/png;base64,{img}"}}]))
+            chat_history.add_message(
+                ChatMessageContent(
+                    role="user",
+                    items=[ImageContent(uri=f"data:image/png;base64,{img}")]
+                )
+            )
+
+    service, req_params = get_completion_service()
+    # Set the response format to JSON object
+    req_params.extension_data["response_format"] = {"type": "json_object"}
```

Comment:
This assumes using a model that supports structured outputs - if this is not the desired route, this will require refactoring.